File: 1100/1557B/main.java
import java.util.*;
public class main
{
    public static void main(String args[]) 
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
 
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int m = sc.nextInt();
 
            TreeMap<Integer,Integer> map = new TreeMap<>();
            for(int j=0;j<n;j++)
            {
                int temp = sc.nextInt();
                map.put(temp,j);
            }
 
            int prev = -2;
            int count = 0;
 
            for(Map.Entry<Integer, Integer> en : map.entrySet()) 
            {
                Integer val = en.getValue();
                if(val - prev ==1)
                    count++;
 
                prev = val;
            }
 
            if(count >= n-m)
                System.out.println("Yes");
            else 
                System.out.println("NO");
        }
 
    }
}

File: 1100/1557B/main.cpp
#include<bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while(t--) 
    {
        int n, m;
        cin >> n >> m;

        map<int, int> mp;
        for(int j = 0; j < n; j++) 
        {
            int temp;
            cin >> temp;

            mp[temp] = j;
        }

        int prev = -2;
        int count = 0;

        for(auto const& [key, val] : mp) 
        {
            if(val - prev == 1)
                count++;

            prev = val;
        }

        if(count >= n - m)
            cout << "YES\n";
        else
            cout << "NO\n";
    }

    return 0;
}